src/snowflake/snowpark/_internal/telemetry.py

```diff
@@ -344,24 +345,51 @@ def send_session_created_telemetry(self, created_by_snowpark: bool):
         }
         self.send(message)
 
-    @safe_telemetry
-    def send_upload_file_perf_telemetry(
-        self, func_name: str, duration: float, sfqid: str
+    def send_performance_telemetry(
```

Comment:
do we also need @safe_telemetry decorator here?

Reply:
oh yes, thanks for calling this out